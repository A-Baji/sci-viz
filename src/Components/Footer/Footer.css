footer {
  display: flex;
  position: fixed;
  bottom: 0;
  width: 100vw;
  background-color: #f1f1f1;
  box-shadow: 0px -3px 3px #c8c9c7;
  padding: 8px 0;
}

.footer-content {
  position: relative;
  height: 100%;
  align-self: center;
  width: 80%;
}

.footer-content p {
  margin: 0 25% 0 55%;
  width: fit-content;
  font-size: 0.9rem;
  text-align: center;
}

.version-info-div {
  position: absolute;
  text-align: right;
  right: 2%;
  font-size: 0.75rem;
  color: #399e5a;
}

.version-number {
  margin-top: 1px;
  margin-bottom: 1px;
}

@media only screen and (max-width: 1366px) {
  .footer-content {
    width: 70%;
  }

  .footer-content p {
    margin: 0 20% 0 50%;
    font-size: 0.85rem;
  }
}

@media only screen and (max-width: 960px) {
  .footer-content {
    width: 60%;
  }

  .footer-content p {
    margin-left: 2%;
    font-size: 0.8rem;
  }
}
