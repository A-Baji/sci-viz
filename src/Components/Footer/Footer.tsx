import React from 'react'
import { version } from '../../../package.json'

import './Footer.css'

interface FooterProps {
    databaseHost?: 'CHETANA'
}

interface FooterState {
    backendVersion: string
}

/**
 * Footer component
 */
export default class Footer extends React.Component<FooterProps, FooterState> {
    constructor(props: FooterProps) {
        super(props)

        this.state = {
            backendVersion: ''
        }
    }

    /**
     * Get the version number upon being mounted.
     */
    componentDidMount() {
        let apiUrl = `${process.env.REACT_APP_DJSCIVIZ_BACKEND_PREFIX}/version`

        if (this.props.databaseHost) {
            apiUrl = apiUrl.concat(`&database_host=${this.props.databaseHost}`)
        }

        fetch(apiUrl, {
            method: 'GET',
            headers: { 'Content-Type': 'application/json' }
        })
            .then((result) => {
                // Check for error mesage 500, if so throw and error
                if (result.status === 500) {
                    throw new Error('Unable to get version number')
                }

                return result.json()
            })
            .then((result) => {
                if (result.version) {
                    this.setState({ backendVersion: result.version })
                }
            })
            .catch((error) => {
                this.setState({ backendVersion: 'Unable to get version number' })
            })
    }

    render() {
        return (
            <footer>
                <div className='footer-content'>
                    <p>Â© 2021, DataJoint SciViz</p>
                </div>
                <div className='version-info-div'>
                    <div className='version-number'>
                        <b>Front End Version:</b> {version} <b>Back End Version:</b>{' '}
                        {this.state.backendVersion}
                    </div>
                </div>
            </footer>
        )
    }
}
