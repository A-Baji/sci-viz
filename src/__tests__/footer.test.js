import React from 'react'
import renderer from 'react-test-renderer'

import Footer from '../Components/Footer/Footer.tsx'

beforeAll(() => {
    process.env = Object.assign(process.env, { NODE_ICU_DATA: 'node_modules/full-icu' })
})

global.fetch = jest.fn(() =>
    Promise.resolve({
        json: () => Promise.resolve({ version: '1.0.2' })
    })
)

it('Snapshot of Footer component', () => {
    const tree = renderer.create(<Footer />).toJSON()
    expect(tree).toMatchSnapshot()
})
