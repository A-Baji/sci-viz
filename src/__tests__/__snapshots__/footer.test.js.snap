// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Footer component 1`] = `
<footer>
  <div
    className="footer-content"
  >
    <p>
      Â© 2023, DataJoint SciViz
    </p>
  </div>
  <div
    className="version-info-div"
  >
    <div
      className="version-number"
    >
      <b>
        Front End Version:
      </b>
       
      2.2.0
       
      <b>
        Back End Version:
      </b>
       
    </div>
  </div>
</footer>
`;
